internal/rootcoord/meta_table.go

```diff
@@ -403,11 +422,13 @@ func (mt *MetaTable) ListCollections(ctx context.Context, ts Timestamp) ([]*mode
 
 func (mt *MetaTable) listCollectionFromCache() ([]*model.Collection, error) {
 	collectionFromCache := make([]*model.Collection, 0)
-	for _, meta := range mt.collID2Meta {
-		if meta.Available() {
-			collectionFromCache = append(collectionFromCache, meta)
+	filterFn := func(k int64, v *model.Collection) bool {
+		if v.Available() {
+			collectionFromCache = append(collectionFromCache, v)
 		}
+		return true
 	}
+	mt.collID2Meta.Scan(filterFn)
```

Comment:
Will `Scan` read all records in catalog?

Reply:
Nope, it just scan the cache data